aws/data_source_aws_guardduty_detector_test.go

```diff
@@ -0,0 +1,38 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccAWSGuarddutyDetectorDataSource_basic(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsGuarddutyDetectorConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair("data.aws_guardduty_detector.test", "id", "aws_guardduty_detector.test", "id"),
+					resource.TestCheckResourceAttr("data.aws_guardduty_detector.test", "enabled", "true"),
+					resource.TestMatchResourceAttr("data.aws_guardduty_detector.test", "service_role_arn", regexp.MustCompile("^arn:aws:iam::[0-9]{12}:role/aws-service-role/guardduty.amazonaws.com/AWSServiceRoleForAmazonGuardDuty")),
+					resource.TestCheckResourceAttr("data.aws_guardduty_detector.test", "finding_publishing_frequency", "SIX_HOURS"),
```

Comment:
We can also check this data source value matches the resource here 👍 

```suggestion
					resource.TestCheckResourceAttrPair("data.aws_guardduty_detector.test", "finding_publishing_frequency", "aws_guardduty_detector.test", "finding_publishing_frequency"),
```

Reply:
All tests have been updated to compare with the resource properties and not hard-coded values.